src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -306,6 +306,12 @@ export class MagicWandService {
 
     // Updates shortest path between neighbor pixel and original node.
     while (toVisit.getSize() !== 0) {
+      // This if statement limits the size of the priority queue for 
+      // reasonable runtime performance in the browser. Removing this 
+      // may cause the browser to become unresponsive for big previews.
+      if (toVisit.getSize() > 600000) {
+        return distances;
+      }
```

Comment:
What does this mean in terms of the solution?  Will the user still be able to change the tolerance and only get a partial of the flood fill?  It's good to include these ideas in your comments.

Reply:
Thanks! Will expand the explanation in my in-line comment